control-plane/p4RuntimeArchStandard.cpp

```diff
@@ -0,0 +1,715 @@
+                    symbols->add(SymbolType::ACTION_PROFILE(), *instance->name);
+                }
+            }
+        }
+        {
+            auto instance = getExternInstanceFromProperty(
+                table,
+                P4V1::V1Model::instance.tableAttributes.counters.name,
+                refMap,
+                typeMap,
+                &isConstructedInPlace);
+            if (instance != boost::none) {
+                if (instance->type->name != P4V1::V1Model::instance.directCounter.name) {
+                    ::error("Expected a direct counter: %1%", instance->expression);
+                } else if (isConstructedInPlace) {
+                    symbols->add(SymbolType::DIRECT_COUNTER(), *instance->name);
+                }
+            }
+        }
+        {
+            auto instance = getExternInstanceFromProperty(
+                table,
+                P4V1::V1Model::instance.tableAttributes.meters.name,
+                refMap,
+                typeMap,
+                &isConstructedInPlace);
+            if (instance != boost::none) {
+                if (instance->type->name != P4V1::V1Model::instance.directMeter.name) {
+                    ::error("Expected a direct meter: %1%", instance->expression);
+                } else if (isConstructedInPlace) {
+                    symbols->add(SymbolType::DIRECT_METER(), *instance->name);
+                }
+            }
+        }
+    }
+
+    void collectExternInstance(P4RuntimeSymbolTableIface* symbols,
+                               const IR::ExternBlock* externBlock) override {
+        CHECK_NULL(externBlock);
+
+        auto decl = externBlock->node->to<IR::IDeclaration>();
+        if (decl == nullptr) return;
+
+        if (externBlock->type->name == P4V1::V1Model::instance.counter.name) {
+            symbols->add(SymbolType::COUNTER(), decl);
+        } else if (externBlock->type->name == P4V1::V1Model::instance.directCounter.name) {
+            symbols->add(SymbolType::DIRECT_COUNTER(), decl);
+        } else if (externBlock->type->name == P4V1::V1Model::instance.meter.name) {
+            symbols->add(SymbolType::METER(), decl);
+        } else if (externBlock->type->name == P4V1::V1Model::instance.directMeter.name) {
+            symbols->add(SymbolType::DIRECT_METER(), decl);
+        } else if (externBlock->type->name == P4V1::V1Model::instance.action_profile.name ||
+                   externBlock->type->name == P4V1::V1Model::instance.action_selector.name) {
+            symbols->add(SymbolType::ACTION_PROFILE(), decl);
+        } else if (externBlock->type->name == P4V1::V1Model::instance.registers.name) {
+            symbols->add(SymbolType::REGISTER(), decl);
+        }
+    }
+
+    void collectExternFunction(P4RuntimeSymbolTableIface* symbols,
+                               const P4::ExternFunction* externFunction) override {
+        auto digest = getDigestCall(externFunction, refMap, typeMap, nullptr);
```

Comment:
How about clone, clone3, resubmit, recirculate? these are all v1model extern functions. No support in P4Runtime yet?

Reply:
Never mind, they are not controllable from the control plane.

Reply:
Clone is configured at runtime and there is support in P4Runtime, but no information is required in P4Info.